Java/core/src/main/java/com/amazon/randomcutforest/tree/AbstractCompactRandomCutTree.java

```diff
@@ -510,6 +510,11 @@ public int getRootIndex() {
         return intValue(root);
     }
 
+    @Override
+    protected boolean referenceEquals(Integer oldPointRef, Integer pointRef) {
+        return oldPointRef.equals(pointRef);
```

Comment:
Would there be any value to adding short-circuiting here? I.e., will the old ref ever be the same as the new ref?
```
return oldPointRef == pointRef || oldPointRef.equals(pointRef);
```

Reply:
There could be some value for classes which are more complicated than Integer. Perhaps we can add it to those classes?